### ISSUE TITLE
Pull request returning error for non existing urls of committed files 

### ISSUE THREAD

**Issue by Delphine-L** [galaxyproject/training-material#546]
Hi,

I report an issue after failed test on unicycler training pull request (https://github.com/galaxyproject/training-material/pull/476). Travis returns errors of non existing external urls (https://travis-ci.org/galaxyproject/training-material/builds/277310665?utm_source=github_status&utm_medium=notification).  The urls at fault are linking to files committed in the pull request (eg: https://github.com/galaxyproject/training-material/tree/master/topics/assembly/tutorials/unicycler-assembly/tutorial.md). The errors don't appear in previous or following commits. Not sure if it is a problem with the test or with how I wrote the links ?

Thanks for the help,

**Comment by martenson** [comment]
The errors:

```
Checking 763 external links...
Ran on 98 files!
- ./_site/topics/assembly/index.html
  *  External link https://galaxyproject.github.io/training-material/topics/assembly/tutorials/unicycler-assembly/slides.html failed: 404 No error
  *  External link https://galaxyproject.github.io/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html failed: 404 No error
  *  External link https://github.com/%7B failed: 404 No error
- ./_site/topics/assembly/tutorials/unicycler-assembly/tutorial.html
  *  External link http://www.nature.com/nmeth/journal/v9/n4/full/nmeth.1935.html failed: 401 No error
  *  External link https://galaxyproject.github.io/training-material/topics/assembly/tutorials/unicycler-assembly/slides failed: 404 No error
  *  External link https://github.com/galaxyproject/training-material/tree/master/topics/assembly/tutorials/unicycler-assembly/tutorial.md failed: 404 No error
htmlproofer 3.7.2 | Error:  HTML-Proofer found 6 failures!
```

**Comment by willdurand** [comment]
This is fine. Most of the PRs won't add new links so it is better not to try to change the behavior that is correct 90% of the time.

You did nothing wrong, and if everything in the PR is ok, the PR will get merged 💪☺️

**Comment by martenson** [comment]
@willdurand Maybe we shouldn't run the tests on the PR but only on master then. Failing tests for unknown reasons (to contributors) are discouraging.

**Comment by willdurand** [comment]
@martenson I disagree. It is *really* important to have tests before merging code into master.

**Comment by bgruening** [comment]
> before merging **code** into

Just nitpicking, but this is kind of important for me and also one reason I see this repo as different then the other repos we usually contribute. Imho this is not about **code** here, and our contributors are not strictly coders. I think this is important to keep in mind - and maybe it justifies to treat this repo in a different way.

**Comment by willdurand** [comment]
Yup. In the meantime, we've discussed a fix on gitter that should fix this issue. I'll give it a try.

**Comment by shiltemann** [comment]
some of these are coming from the topic template (probably you added the metadata in that commit so these links were automatically made by the template at that time) ..but we could remove `{{ site.url }}` prefix from links in the template and make them relative links so that they are checked against the locally built version of the site, not the online one. That should take care of 3 of the errors 
 
And this error: `External link https://github.com/%7B failed: 404 No error` will be fixed by changing the metadata.yaml to use the new convention of adding just usernames for contributers/maintainers as Will remarked. Nature.com links just need to be replaced with open source views of the articles.

The "edit me on github" links to the repo directly will be harder to solve, but that gets us down to 1 broken link here only no?  and htmlproofer has an option `--url_swap` that looks like we could maybe use to rewrite these links to match the fork/branch the PR is made in to see if it exists there? (if we have access to that information in travis)

**Comment by bebatut** [comment]
It should be fixed by now with #557
